compiler/src/typed/typecore.re

```diff
@@ -2866,27 +2997,40 @@ let report_error = (env, ppf) =>
         );
       };
     }
-  | Too_many_arguments(in_function, ty, explanation) => {
+  | Not_a_function(ty, explanation) => {
       reset_and_mark_loops(ty);
-      if (in_function) {
-        fprintf(ppf, "This function expects too many arguments,@ ");
+      fprintf(ppf, "This expression is not a function,@ ");
+      fprintf(
+        ppf,
+        "the expected type is@ %a%t",
+        type_expr,
+        ty,
+        report_type_expected_explanation_opt(explanation),
+      );
+    }
+  | Function_label_mismatch({got, expected, expected_type, explanation}) => {
+      reset_and_mark_loops(expected_type);
+      if (is_optional(got)) {
         fprintf(
           ppf,
-          "it should have type@ %a%t",
-          type_expr,
-          ty,
-          report_type_expected_explanation_opt(explanation),
+          "This function contains the optional, default argument %s@ ",
```

Comment:
```suggestion
          "This function contains the labeled, defaulted argument %s@ ",
```

Reply:
Not a huge fan of `defaulted`. `labeled` here also doesn't feel great because everything is implicitly labeled. I think we should just say "has a default value."